### ISSUE TITLE
There is a difference between `[]` and `None`

### ISSUE THREAD

**Issue by matantsu** [GoogleCloudPlatform/endpoints-proto-datastore#138]
I have a Model:
```
class foo(EndpointsModel):
    bar = NDB.KeyProperty(kind='baz', repeated=True)

@foo.method(...)
def update(self, f):
   f.put()
   return f
```

when I use the API explorer and enter `{ bar: [...]}`everything works and `bar` is set correctly.

but when I do not set `bar` in the request , `bar` is implicitly set to `[]` and my previous data is lost.

I would like the data to be lost only when I explicitly set `{bar: []}`

**Comment by dhermes** [comment]
Sorry I was traveling internationally when you filed this. Great point!

**Comment by dhermes** [comment]
@matantsu I fixed in #140.

Note that the semantics of PATCH and PUT (patch vs. update) are a bit subtle and to actually get PATCH semantics, you'd need to use

```python
@MyModel.method(request_message=MyModel.ProtoModel(), ...)
```

and then directly update the entire `ndb` entity from the ProtoRPC message.
